### PR TITLE
HTTP client that acts more like curl.

### PR DESCRIPTION
**Pull request by alexaryn**
Example use:
```
c = OneShotKaClient("https://test-api.foo.com/v1/magic")
tok = "ey...AQ"
hdr = {"Authorization": f"Bearer {tok}"}
form = {
    "options": '{"use_foo": true}',
    "pdf": "@files/doc.pdf",
}
buf = c.post(headers=hdr, form=form)
print(buf)
print(c.resp.status, c.resp.getheaders())
```

### CODE REVIEW

File: lib/sycamore/sycamore/utils/http.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            "Accept": "*/*",
+            "Accept-Encoding": "identity",
+        }
+        hdr.update(headers)
+
+        self._htconn()
+        self.conn.request("GET", self.up.path, headers=hdr)
+        return self._finish()
+
+    def post(self, *, headers: dict[str, str] = {}, form: dict[str, str] = {}) -> bytes:
```
Comment (MarkLindblad) [inline lib/sycamore/sycamore/utils/http.py:40]: I think you narrowly miss [this issue](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python) by how carefully the code is written. No need for a change.

Comment (MarkLindblad) [reply lib/sycamore/sycamore/utils/http.py:40]: I am not sure how this would affect multithreading.

**Review by MarkLindblad — APPROVED** [review]
Beautiful code that looks easy to use.

File: lib/sycamore/sycamore/utils/http.py (hunk 1 above)
Comment (alexaryn) [reply lib/sycamore/sycamore/utils/http.py:40]: I was debating defaulting to None.  Perhaps it's a better policy for the future.

I don't think threading is an issue for this class, as there's no real sharing.  One instance of OneShotKaClient can be passed from thread to thread as long as only one thread uses it at a time.

Comment (MarkLindblad) [reply lib/sycamore/sycamore/utils/http.py:40]: My concern is that all instances are sharing the same default dictionary and that they need to be careful to take turns changing it.

**Pull request merged**